packages/service-library/src/servicelib/rabbitmq.py

```diff
@@ -93,19 +46,75 @@ class RabbitMQClient:
     _rpc_channel: aio_pika.abc.AbstractChannel | None = None
     _rpc: RPC | None = None
 
-    def __post_init__(self):
+    _healthy_state: bool = True
+
+    def __post_init__(self) -> None:
         # recommendations are 1 connection per process
         self._connection_pool = aio_pika.pool.Pool(
-            _get_connection, self.settings.dsn, self.client_name, max_size=1
+            self._get_connection, self.settings.dsn, self.client_name, max_size=1
         )
         # channels are not thread safe, what about python?
         self._channel_pool = aio_pika.pool.Pool(self._get_channel, max_size=10)
 
+    def _connection_close_callback(
+        self,
+        sender: Any,  # pylint: disable=unused-argument
+        exc: BaseException | None,
+    ) -> None:
+        if exc:
+            if isinstance(exc, asyncio.CancelledError):
+                _logger.info("Rabbit connection cancelled")
+            elif isinstance(exc, aiormq.exceptions.ConnectionClosed):
+                _logger.info("Rabbit connection closed: %s", exc)
+            else:
+                _logger.error(
+                    "Rabbit connection closed with exception from %s:%s",
+                    type(exc),
+                    exc,
+                )
+                self._healthy_state = True
```

Comment:
Question: I don't understand why it is considered healthy when the connection closes with an error?

Reply:
Also I only see this set to True. I don't understand if this does anything.

Reply:
very good catch... that is due to some renaming from @pcrespov review. I forgot to change the value. very good. thanks